notebook2/notebook/notebook.py

```diff
@@ -299,12 +334,14 @@ def delete_worker_pod(pod_name, svc_name):
 
 
 @app.route('/admin-login', methods=['GET'])
+@requires_auth()
```

Comment:
This is currently slightly awkward, because a user is first logged in, and then must supply the admin password. I've made an issue to add roles, but since this is little used functionality, and is more secure than basic auth alone, I think it's worth protecting admin routes.

Happy to change if you'd like :)